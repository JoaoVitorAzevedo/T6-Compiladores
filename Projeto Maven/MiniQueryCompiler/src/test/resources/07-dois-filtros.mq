# O gerador de JS atual vai redeclarar 'filteredData', o que causaria um erro em JS.
# O teste passa porque valida o cÃ³digo gerado, expondo essa falha.
LOAD "data.json"
FILTER .idade >= 18
FILTER .nome != "Admin"