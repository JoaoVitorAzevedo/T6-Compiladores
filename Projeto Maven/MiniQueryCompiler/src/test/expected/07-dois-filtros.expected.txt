ANÁLISE SEMÂNTICA VÁLIDA

CÓDIGO JS GERADO:
// Código gerado automaticamente
const fs = require('fs');

const input = JSON.parse(fs.readFileSync('data.json', 'utf-8'));
const filteredData = input.filter(item => item.idade >= 18);
filteredData = filteredData.filter(item => item.nome !== "Admin");